src/db/users/audit-session.schema.ts

```diff
@@ -0,0 +1,9 @@
+import { Schema } from "mongoose";
+
+import { AuditSession } from './models/audit-session.model';
+
+export const AuditSessionSchema = new Schema<AuditSession>({
+    _id: {type: String, required: [true, `You must provide a sessionId.`]},
+    expires: {type: Date, required: [true, `You must provide an expiration date.`]},
+    createdAt: {type: Date, default: Date.now()},
+});
```

Comment:
I tried to set up pre-save hooks to do the hashing and creation dates the easy way, but it seems like they don't fire when adding a sub-document to a parent document.